src/components/Modal/index.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import '../../styles/modal.scss';
+
+class Modal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isModalActive: props.show
+        };
+
+        this.toggleModal = this.toggleModal.bind(this);
+    }
+
+    toggleModal(isModalActive) {
+        this.setState({isModalActive});
+
+        if (typeof this.props.onModalClose === 'function') {
+            this.props.onModalClose();
+        }
+    }
+
+    render() {
+        return this.state.isModalActive ? (
```

Comment:
it should be only representative component, it should be decided outside whether to open or close the modal

Reply:
Fixed, thank you!